Core/MIPS/MIPSAnalyst.cpp

```diff
@@ -1066,14 +1041,32 @@ namespace MIPSAnalyst {
 			if (end) {
 				currentFunction.end = addr + 4;
 				currentFunction.isStraightLeaf = isStraightLeaf;
+
+				// Check if we already have symbol info starting here.  If so, skip insertion.
+				// We used to use the symbols to find the functions, but sometimes we'd find
+				// wrong ones due to two modules with the same name.
+				u32 existingSize = g_symbolMap->GetFunctionSize(currentFunction.start);
+				if (existingSize != SymbolMap::INVALID_ADDRESS) {
+					currentFunction.foundInSymbolMap = true;
+
+					// If we run into a func with a different size, skip updating the hash map.
+					// This will prevent us saving incorrectly named funcs with wrong hashes.
```

Comment:
well, at least if we're lucky enough that the sizes actually are different :) But that should be most of the time.